### PR TITLE
Updated list of files in service worker for latest Webviewer version

### PR DESCRIPTION
**Pull request by Johnny-kung**
Update the code to adapt the latest version of Webviewer.

Issue:
WVR-1186

### CODE REVIEW

**Review by mparizeau-pdftron — CHANGES_REQUESTED** [review]
Loading PDF doesn't work for me. Looks like it's missing the files in the optimized folder pdf/lean/optimized

**Comment by Johnny-kung** [comment]
I added a script to generate a json file, which contains the list of the files' path to be cached by service worker.

File: prepare-serviceworker-list.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+const path = require('path');
+
+const outputPath = './src/service-worker-list.json';
+
+// According to the need, add or remove directories from this array
+const targetFolders = [
+  { name: 'core', path: './public/lib/core/' },
```
Comment (mparizeau-pdftron) [inline prepare-serviceworker-list.js]: Looks like it doesn't work if I try to content editing. Not sure if we need to explicitly specify `contentEdit`, `officeEditor` and `assets` folders?

**Review by mparizeau-pdftron — CHANGES_REQUESTED** [review]

**Comment by Johnny-kung** [comment]
Screen Shot 2023-10-12 at 11 04 19 AM
Screen Shot 2023-10-12 at 11 04 36 AM

Content edit seems not able to edit the first page in the sample pdf file, either with the server running or in offline. This is probably not related to offline case itself.

**Comment by Johnny-kung** [comment]
Confirmed with Jack that it's a bug related to content edit feature, not specific in this sample.

**Review by mparizeau-pdftron — APPROVED** [review]

**Pull request merged**